l10n_nl_bank/migrations/13.0.1.1.0/post-migration.py

```diff
@@ -0,0 +1,24 @@
+# 2021 Bosd
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""
+The banks have been created with the no update tag.
+Use openupgradelib to force load the changes.
+Old entries will be unchanged,
+new entries will be added and existing entries will be updated.
+"""
+
+
+import logging
+
+_logger = logging.getLogger(__name__)
+try:
+    from openupgradelib import openupgrade
+except ImportError:
+    _logger.warning("OpenUpgradeLib is not found, can't update dutch bank data")
+
+
+@openupgrade.migrate(use_env=True)
```

Comment:
Because openupgrade might not be available. you are not allowed to refer to its decorator in this way. No problem, because the method only uses the cursor which is what you get without the decorator instead of the env.

Reply:
That makes sense. What is the properway to solve this?

Reply:
Remove the decorator and change signature to receive the cr that Odoo passes by default (https://github.com/OCA/l10n-netherlands/blob/11.0/l10n_nl_tax_statement/migrations/11.0.2.0.0/post-migrate.py#L8)

Reply:
Can't get that to work. It keeps throwing server errors :(

```
def migrate(cr, version):

    openupgrade.load_data(env.cr, "l10n_nl_bank", "data/res_bank_data.xml", mode="init")
```

Reply:
What kind of errors exactly?

Reply:
Might want to try passing `cr` to load_data instead of `env.cr` in that case.

Reply:
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 624, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/odoo/custom/src/odoo/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 669, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 350, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 915, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/custom/src/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/auto/addons/web/controllers/main.py", line 1331, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/opt/odoo/auto/addons/web/controllers/main.py", line 1319, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 387, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/api.py", line 374, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-61>", line 2, in button_immediate_upgrade
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_module.py", line 72, in check_and_log
    return method(self, *args, **kwargs)
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_module.py", line 634, in button_immediate_upgrade
    return self._button_immediate_function(type(self).button_upgrade)
  File "/opt/odoo/custom/src/odoo/odoo/addons/base/models/ir_module.py", line 573, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/opt/odoo/custom/src/odoo/odoo/modules/registry.py", line 86, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 419, in load_modules
    force, status, report, loaded_modules, update_module, models_to_check)
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 315, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/opt/odoo/custom/src/odoo/odoo/modules/loading.py", line 230, in load_module_graph
    migrations.migrate_module(package, 'post')
  File "/opt/odoo/custom/src/odoo/odoo/modules/migration.py", line 180, in migrate_module
    migrate(self.cr, installed_version)
  File "/opt/odoo/auto/addons/l10n_nl_bank/migrations/13.0.1.1.0/post-migration.py", line 23, in migrate
    openupgrade.load_data(env.cr, "l10n_nl_bank", "data/res_bank_data.xml", mode="init")
NameError: name 'openupgrade' is not defined

Reply:
Ah yes of course. Try something like

```
try:
    from openupgradelib import openupgrade
except ImportError:
    openupgrade = None

def migrate(cr, version):
    if openupgrade is None:
        _logger.warning("OpenUpgradeLib is not found, can't update dutch bank data")
        return
    ...
```

Reply:
Maybe you noticed I don't understand exactly what we are doing here. :)
But the version without the requested change was working in both cases. While openupgrade lib was available and when it was unavailable.

Reply:
Thank you for your patience and your effort!

Reply:
Great to be part of this community and learn a lot!
Changes are in the latest commit.

Reply:
Perfect! Can you squash commits?

Reply:
Done! :+1: